### ISSUE TITLE
Evaluate if DKIM can/should be implemented

### ISSUE THREAD

**Issue by wneessen** [wneessen/go-mail#28]
### Is your feature request related to a problem? Please describe.

DKIM might be a feature a proper mail library should support. Of course there are already DKIM implementations out there, but that would break our "only standard library" premise. We might wanna evaluate if DKIM can be easily implemented as part of this library.

### Describe the solution you'd like

Solution should allow signing mails with domain key headers.

**Comment by inliquid** [comment]
I'm wondering what is the status at the moment? Any conclusion whether it's feasible / easy enough ti implement to become a feature?

**Comment by wneessen** [comment]
I am still working through the RFC, but some solid groundwork has already been done. I expect it to be done by end of this month, if I find enough free time.

**Comment by wneessen** [comment]
After spending some time implementing DKIM myself, I decided to cancel the implementation directly into go-mail. Since we now have `Middleware` we have a much more elegant tool at hand to implement DKIM with go-mail.

Therefore I've started work on a DKIM middleware instead in the go-mail-middleware repository. I have a working proof of concept in https://github.com/wneessen/go-mail-middleware/tree/feature/dkim/dkim and just need to perform some fine-tuneing, add tests and documentation. I expect this to go live in the next days.

Therefore I will close this feature-request in favour of the middlware.
